components/application-operator/pkg/kymahelm/application/overrides.go

```diff
@@ -13,3 +23,52 @@ type OverridesData struct {
 	StrictMode                            string `json:"strictMode,omitempty"`
 	IsBEBEnabled                          bool   `json:"isBEBEnabled,omitempty"`
 }
+
+type OverridesMap map[string]interface{}
+
+type StringMap map[string]string
+
+func MergeLabelOverrides(labels StringMap, target map[string]interface{}) {
```

Comment:
The code could be a bit shorter:
```
func MergeLabelOverrides(labels StringMap, target map[string]interface{}) {
	for key, value := range labels {
		if strings.HasPrefix(key, overridePrefix) {
			preParsedKey := strings.Trim(strings.TrimPrefix(key, overridePrefix),".")

			if len(preParsedKey) == 0 {
				continue
			}

			utils.MergeMaps(target, unwind(preParsedKey, value))
		}
	}
}

func unwind(key string, value string) map[string]interface{} {
	index := strings.Index(key, ".")
	if index < 0 {
		return map[string]interface{}{key: value}
	}

	return map[string]interface{}{key[:index]: unwind(key[index + 1:], value)}
}
```

Moreover it more nicely communicates the recursive nature of the unwind function ( stop conditions are placed first, and recursive call is the last thing in the function body)

Reply:
Corrected.